src/pudl/transform/ferc1.py

```diff
@@ -3546,6 +3638,64 @@ def process_xbrl_metadata(self, xbrl_metadata_json) -> pd.DataFrame:
         return pd.concat([meta, ending_balance])
 
 
+class ElectricitySalesByRateScheduleFerc1TableTransformer(
+    Ferc1AbstractTableTransformer
+):
+    """Transform class for :ref:`electricity_sales_by_rate_schedule_ferc1` table."""
+
+    table_id: TableIdFerc1 = TableIdFerc1.ELECTRICITY_SALES_BY_RATE_SCHEDULE_FERC1
+    has_unique_record_ids: bool = False
+
+    def add_axis_to_total_table_rows(self, df: pd.DataFrame):
+        """Add total to the axis column for rows from the total table.
+
+        Because we're adding the
+        sales_of_electricity_by_rate_schedules_account_totals_304 table into the mix,
+        we have a bunch of total values that get mixed in with all the _billed columns
+        from the individual tables. If left alone, these totals aren't labeled in any
+        way becuse they don't have the same _axis columns explaining what each of the
+        values are. In order to distinguish them from the rest of the sub-total data we
+        use this function to create an _axis value for them noting that they are totals.
+
+        It's worth noting that there are also some total values in there already.
+        Those would be hard to clean. The idea is that if you want the actual totals,
+        don't try and sum the sub-components, look at the actual labeled total rows.
+
+        This function relies on the ``sched_table_name`` column, so it must be called
+        before that gets dropped.
+
+        Args:
+            df: The sales table with a ``sched_table_name`` column.
+        """
+        logger.info(f"{self.table_id.value}: Labeling total values.")
+        df.loc[
+            df["sched_table_name"]
+            == "sales_of_electricity_by_rate_schedules_account_totals_304",
+            ["sales_axis", "rate_schedule_description"],
+        ] = "total"
+        return df
+
+    @cache_df(key="xbrl")
+    def process_xbrl(
+        self,
+        raw_xbrl_instant: pd.DataFrame,
+        raw_xbrl_duration: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Rename columns before running wide_to_tidy."""
+        logger.info(f"{self.table_id.value}: Processing XBRL data pre-concatenation.")
+        return (
+            self.merge_instant_and_duration_tables_xbrl(
+                raw_xbrl_instant, raw_xbrl_duration
+            )
+            .pipe(self.rename_columns, rename_stage="xbrl")
+            .pipe(self.combine_axis_columns_xbrl)
+            .pipe(self.add_axis_to_total_table_rows)
+            .pipe(self.wide_to_tidy, source_ferc1=SourceFerc1.XBRL)
+            .pipe(self.assign_record_id, source_ferc1=SourceFerc1.XBRL)
+            .pipe(self.assign_utility_id_ferc1, source_ferc1=SourceFerc1.XBRL)
+        )
```

Comment:
this is not blocking, but if you wanted instead of overwriting the whole `process_xbrl` you could overwrite just `combine_axis_columns_xbrl` with a:
```
    def combine_axis_columns_xbrl(
        self, df: pd.DataFrame, params: CombineAxisColumnsXbrl | None = None
    ) -> pd.DataFrame:
        """Docs!"""
        return super().combine_axis_columns_xbrl(df, params).pipe(self.add_axis_to_total_table_rows)
```
 These two feel thematically similar enough and it would avoid any changes in the parent `process_xbrl` not propagating down here.

Reply:
I overwrote the `process_xbrl` function mostly because I needed to `rename_columns` to come before `wide_to_tidy`